android/src/com/mapswithme/util/sharing/ViralEditorShareable.java

```diff
@@ -0,0 +1,81 @@
+package com.mapswithme.util.sharing;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.net.Uri;
+import android.support.annotation.DrawableRes;
+import android.support.annotation.Nullable;
+
+import java.util.Locale;
+
+import com.facebook.FacebookSdk;
+import com.facebook.share.model.ShareLinkContent;
+import com.facebook.share.widget.ShareDialog;
+import com.mapswithme.util.UiUtils;
+
+public class ViralEditorShareable extends BaseShareable
+{
+  private static final String FACEBOOK_SHARE_URL = "http://maps.me/fb-editor-v1?lang=" + Locale.getDefault().getLanguage();
+
+  private final Uri mUri;
+
+  public ViralEditorShareable(Activity context, @DrawableRes int resId)
+  {
+    super(context);
+    mUri = UiUtils.getUriToResId(context, resId);
+  }
+
+  @Override
+  protected void modifyIntent(Intent intent, @Nullable SharingTarget target)
+  {
+    super.modifyIntent(intent, target);
+    intent.putExtra(Intent.EXTRA_STREAM, mUri);
+  }
+
+  @Override
+  public String getMimeType()
+  {
+    return TargetUtils.TYPE_TEXT_PLAIN;
+  }
+
+  @Override
+  public void share(SharingTarget target)
+  {
+    Intent intent = getTargetIntent(target);
+    String lowerCaseName = target.activityName.toLowerCase();
+
+    if (lowerCaseName.contains("facebook"))
+    {
+      shareFacebook();
+      return;
+    }
+
+    if (lowerCaseName.contains("sms") || lowerCaseName.contains("mms"))
```

Comment:
Проверю, вероятно. Хотя не думаю что на практике будет хоть какая-то разница.

> On Mar 28, 2016, at 4:08 PM, Alexander Marchuk <notifications@github.com> wrote:
> 
> In android/src/com/mapswithme/util/sharing/ViralEditorShareable.java <https://github.com/mapsme/omim/pull/2568#discussion_r57569440>:
> 
> > +    return TargetUtils.TYPE_TEXT_PLAIN;
> > +  }
> > +
> > +  @Override
> > +  public void share(SharingTarget target)
> > +  {
> > +    Intent intent = getTargetIntent(target);
> > +    String lowerCaseName = target.activityName.toLowerCase();
> > +
> > +    if (lowerCaseName.contains("facebook"))
> > +    {
> > +      shareFacebook();
> > +      return;
> > +    }
> > +
> > +    if (lowerCaseName.contains("sms") || lowerCaseName.contains("mms"))
> Что-то так себе эвристика. Может, на основе target.packageName будет надёжнее?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2568/files/f2dc591f49716076ca1919ceee98db764511cb08#r57569440>